### PR TITLE
Try to fix lz4 compression

### LINKED ISSUE #31421: lz4 compresssion doesn't work with some formats
**Describe what's wrong**

Writing and reading data with lz4 compression doesn't work with some formats (`Native`, `Values`, `JSONCompactEachRow`, `TSKV` and `Arrow`)

**Does it reproduce on recent release?**

Always reproduces

**How to reproduce**

```sql
DROP TABLE IF EXISTS file;
CREATE TABLE file (x UInt64) ENGINE = File('Values', 'data.lz4');
TRUNCATE TABLE file;
INSERT INTO file SELECT * FROM numbers(100000);
SELECT max(x) FROM file;
DROP TABLE file
```

**Expected behavior**

 Writing and reading data with lz4 compression should work with all formats.

**Error message and/or stacktrace**

For `Values` format:
```
Received exception from server (version 21.12.1):
Code: 27. DB::Exception: Received from localhost:9000. DB::Exception: Cannot parse input: expected '(' before: '\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0':  at row 67667: While executing ValuesBlockInputFormat: While executing File. (CANNOT_PARSE_INPUT_ASSERTION_FAILED)
(query: SELECT x FROM file)
```

For `Native` format it outputs garbage instead of `99999`

For `JSONCOmpactEachRow` format:
```
Received exception from server (version 21.10.3):
Code: 27. DB::Exception: Received from localhost:9000. DB::Exception: Cannot parse input: expected ']' before: '56524"]\n["56525"]\n["56526"]\n["56527"]\n["56528"]\n["56529"]\n["56530"]\n["56531"]\n["56532"]\n["56533"]\n["56534"]\n["56535"]\n["56536"]\n["56537"]\n["56538"]\n["56539"]\n["': While executing JSONCompactEachRowRowInputFormat: While executing File. (CANNOT_PARSE_INPUT_ASSERTION_FAILED)
```
For `TSKV` format:
```
Received exception from server (version 21.10.3):
Code: 27. DB::Exception: Received from localhost:9000. DB::Exception: Found garbage after field in TSKV format: x: (at row 50695)
: While executing ParallelParsingBlockInputFormat: While executing File. (CANNOT_PARSE_INPUT_ASSERTION_FAILED)
```

For `Arrow` format:
```
eceived exception from server (version 21.10.3):
Code: 1002. DB::Exception: Received from localhost:9000. DB::Exception: Error while opening a table: Invalid: Not an Arrow file: While executing ArrowBlockInputFormat: While executing File. (UNKNOWN_EXCEPTION)
```

### PR DESCRIPTION
**Pull request by Avogar**
Changelog category (leave one):
- Bug Fix (user-visible misbehaviour in official stable or prestable release)

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fix lz4 compression for output. Closes https://github.com/ClickHouse/ClickHouse/issues/31421

Detailed description / Documentation draft:
...

### CODE REVIEW

**Comment by Avogar** [comment]
I'm not 100% sure about this fix. I didn't fully understand what the problem was and found the fix just by experimenting, but I cannot reproduce bug anymore

**Comment by Avogar** [comment]
No, it doesn't work(

**Comment by Avogar** [comment]
@Mergifyo update

**Comment by Avogar** [comment]
@Mergifyio update

File: src/IO/Lz4DeflatingWriteBuffer.cpp [hunk 1]
```diff
@@ -125,6 +125,8 @@ void Lz4DeflatingWriteBuffer::nextImpl()
         out->position() = out->buffer().begin();
         throw;
     }
+    out->next();
```
Comment (evillique) [inline src/IO/Lz4DeflatingWriteBuffer.cpp:128]: These two lines seem a little unnecessary.

**Review by evillique — APPROVED** [review]
Turning `autoflash` on together with the updated lz4 library does all the work. The result file size is not affected, and it is actually a great way to fix the problem :)

File: src/IO/Lz4DeflatingWriteBuffer.cpp [hunk 2]
```diff
@@ -125,6 +125,8 @@ void Lz4DeflatingWriteBuffer::nextImpl()
...
+    out_capacity = out->buffer().end() - out->position();
```
Comment (evillique) [inline src/IO/Lz4DeflatingWriteBuffer.cpp:129]: These two lines though seem a little unnecessary.
```suggestion
```

**Comment by Avogar** [comment]
@Mergifyio update

File: src/IO/Lz4DeflatingWriteBuffer.cpp (hunk 2 above)
Comment (Avogar) [reply src/IO/Lz4DeflatingWriteBuffer.cpp:129]: If I remember correctly, if I remove these lines, it will break test. Will check it now

Comment (Avogar) [reply src/IO/Lz4DeflatingWriteBuffer.cpp:129]: Yes, removing of these lines breaks it...

**Comment by Avogar** [comment]
@evillique Can we merge it? Or maybe you want to check smth else

**Comment by evillique** [comment]
Checked some more things, and it works perfectly. The only thing is that Stateless tests flaky check (address, actions) fail because the test takes too much time.

**Comment by Avogar** [comment]
Failed flacky check is ok, it won't be run in master

**Pull request merged**